### PR TITLE
cosmetics: migrate to f-strings

### LINKED ISSUE #54: Minimal python version and f-strings
What is the minimal supported python version?
Can we use f-strings? It would be great if yes.

### PR DESCRIPTION
**Pull request by maratori**
Closes #54

### CODE REVIEW

**Comment by fscherf** [comment]
@maratori Wow thanks! Great changes, only some minor nitpicking from my side :)

File: lona/html/attribute_dict.py [hunk 1]
```diff
@@ -71,13 +71,16 @@ def __getitem__(self, name):
 
     def __setitem__(self, name, value, issuer=None):
         if not isinstance(value, (int, bool, float, str)):
-            raise ValueError('unsupported type: {}'.format(type(value)))
+            raise ValueError(f'unsupported type: {type(value)}')
 
-        if name in ('id', 'class', 'style'):
```
Comment (fscherf) [inline lona/html/attribute_dict.py:76]: Great change, but this is a functional change. I would like a dedicated commit with commit message for it (just append it to this pr)

File: lona/html/attribute_dict.py [hunk 2]
```diff
@@ -124,15 +127,18 @@ def _reset(self, value):
         if not isinstance(value, dict):
             raise ValueError('unsupported type')
 
-        for k, v in value.items():
+        for name, v in value.items():
             if not isinstance(v, (int, bool, float, str)):
                 raise ValueError('unsupported type')
 
-            if k in ('id', 'class', 'style'):
```
Comment (fscherf) [inline lona/html/attribute_dict.py:131]: This is also a funtional change

**Review by fscherf — CHANGES_REQUESTED** [review]

**Comment by fscherf** [comment]
@maratori are f-strings a good idea for logging? My understanding of the ``%s`` Syntax in logging is that it does not use python format strings on purpose to have a unformatted string to fall back to if the formatting fails.

**Comment by maratori** [comment]
@fscherf I'm not sure about specific case when formatting fails.
But you are right (new info for me), that %-formatting is better for logging.
Here is a good article: https://blog.pilosus.org/posts/2020/01/24/python-f-strings-in-logging/.

File: lona/html/attribute_dict.py (hunk 1 above)
Comment (maratori) [reply lona/html/attribute_dict.py:76]: Fixed

File: lona/html/attribute_dict.py (hunk 2 above)
Comment (maratori) [reply lona/html/attribute_dict.py:131]: Fixed

**Review by fscherf — APPROVED** [review]
Looks nicer now and everything seems to work like expected

**Pull request merged**